server/controllers/list.js

```diff
@@ -1794,11 +1794,22 @@ const getAvailableLists = (req, resp) => {
     isArchived: false
   };
 
-  List.find(query, 'name')
-    .lean()
-    .exec()
-    .then(docs => (docs ? resp.send(docs) : resp.sendStatus(400)))
-    .catch(() => resp.sendStatus(400));
+  try {
+    const lists = await List.find(query, 'name')
+      .lean()
+      .populate('cohortId', 'isArchived')
+      .exec();
+
+    const listToSend = lists
+      .filter(
+        list => !list.cohortId || (list.cohortId && !list.cohortId.isArchived)
```

Comment:
`list.cohortId.isArchived`? Shouldn't you populate the `cohort` field instead (based on the `cohortId` field)?

Reply:
we have cohortId field in list schema. when we populate cohort data it is stored in cohortId.